src/libraries/Common/src/Interop/Windows/Kernel32/Interop.CloseHandle.cs

```diff
@@ -8,12 +8,9 @@ internal static partial class Interop
 {
     internal static partial class Kernel32
     {
-#if DLLIMPORTGENERATOR_ENABLED
-        [GeneratedDllImport(Libraries.Kernel32, SetLastError = true)]
-        internal static partial bool CloseHandle(IntPtr handle);
-#else
+#pragma warning disable DLLIMPORTGENANALYZER015 // Use 'GeneratedDllImportAttribute' instead of 'DllImportAttribute' to generate P/Invoke marshalling code at compile time
+                                                // Disabled since CloseHandle is a QCall in some scenarios and DllImportGenerator doesn't support QCalls.
```

Comment:
Why it does not support QCalls?

Reply:
The inner P/Invoke is emitted as a local function to avoid issues with different managed P/Invoke signatures being mapped to the same native signature with different marshalling info. Local functions have compiler-generated names, so we can't rely on the method names in the QCall hookup macros in `ecalllist.h`.

See https://github.com/dotnet/runtimelab/issues/1064, https://github.com/dotnet/runtimelab/pull/1081, and https://github.com/dotnet/runtimelab/pull/1090.

I've proposed within the team creating a QCall-specific generator for usage within System.Private.CoreLib only that uses a different attribute (named QCallAttribute) that doesn't take the module name or EntryPoint name (as the module name is always "QCall" and the EntryPoint name is unused in the QCall loading system) and generates a non-local-function `DllImport` method with a well-defined name in metadata based on the original type and method names. Since QCalls don't support overloading to my knowledge, we won't end up with the overloading problem described in https://github.com/dotnet/runtimelab/issues/1064.

Reply:
I am not sure whether special casing QCalls is worth it. I think it would be better to make the QCalls look more like `System.*.Native` that does not require any special casing.

Reply:
If QCalls had some mechanism to respect the `EntryPoint` metadata from the `DllImport` pseudo-attribute (instead of just using the method name), then we would be able to use DllImportGenerator as-is with `QCalls`.

Right now, the `EntryPoint` metadata is just silently ignored, which makes solving this issue more difficult.

Reply:
We can fix the runtime to respect `EntryPoint`.

I would help with sharing QCall between runtimes. Mono AOT (and NativeAOT) compilers have to be able to find the name of the C entrypoint of the FCall/QCall by looking at the IL, so that they can generate direct call. For FCalls, Mono generates the C entrypoint name by convention from the IL name and signature. Embedding a custom mapping table between the IL names and C entrypoints into the AOT compiler is pain. And having the custom mapping table in the runtime is problematic for static linking.

Switching to EntryPoint as the key for QCalls would help with CoreLib sharing between runtimes.

Reply:
Filed https://github.com/dotnet/runtime/issues/59823.